corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -581,6 +582,17 @@ def soft_undelete_forms(domain, form_ids):
 
         return return_value
 
+    @staticmethod
+    def modify_attachment_xml_and_metadata(form_data, form_attachment_new_xml):
+        attachment_metadata = form_data.get_attachment_meta("form.xml")
+        # Write the new xml to the database
+        XFormAttachmentSQL.write_content(attachment_metadata, form_attachment_new_xml)
```

Comment:
why not `attachment_metadata.write_content(form_attachment_new_xml)`

Reply:
True, fixed here:
https://github.com/dimagi/commcare-hq/pull/19979/commits/d42272370e58270d60fc2b82815e0ad48fdab195